src/infrastructure/services/blockstack-service.ts

```diff
@@ -94,4 +110,132 @@ export class BlockstackService {
             return registeredDomainArray[0];
         }
     }
+    public getAddressMap = async (blockstackID: BlockstackId, tag?: string): Promise<IAddressMapping> => {
+        const cruxPayFileName = CruxSpec.blockstack.getCruxPayFilename(blockstackID);
+        return getContentFromGaiaHub(blockstackID.toString(), cruxPayFileName, this.bnsNodes, tag, this.cacheStorage);
+    }
+    public putAddressMap = async (addressMapping: IAddressMapping, cruxDomainId: CruxDomainId, keyManager: IKeyManager): Promise<string> => {
+        const blockstackID = await this.getBlockstackIdFromKeyManager(keyManager, cruxDomainId);
+        if (!blockstackID) {
+            throw errors.ErrorHelper.getPackageError(null, errors.PackageErrorCode.UserDoesNotExist);
+        }
+        const addressDecoder: Decoder<IAddress> = object({
+            addressHash: string(),
+            secIdentifier: optional(string()),
+        });
+        try {
+            for ( const currency of Object.keys(addressMapping) ) {
+                const addressObject: IAddress = addressDecoder.runWithException(addressMapping[currency]);
+            }
+        } catch (error) {
+            throw ErrorHelper.getPackageError(error, PackageErrorCode.AddressMappingDecodingFailure);
+        }
```

Comment:
this piece of code belongs to Validations class. And the validation of addressMap is not blockstack-service's concern.

Reply:
Moved to Crux Spec validations and removed from blockstack service